### PR TITLE
Github Workflow v2.1

### PR DESCRIPTION
**Pull request by gpmayorga**
This PR fixes:
- Concurrency between jobs
- Deployment logic for concurrent jobs competing to run functions
- Better security setup for GH Actions following guidelines in https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions
- Run deploy-functions only when necessary
- Fix cleanup-pr (tested by closing and reopening PR)
- Add documentation and comments
- Implement a deploy strategy that will make it easier to create new environments or deployments as a separate action

Pending: test production deployments after merge

### CODE REVIEW

**Comment by gpmayorga** [comment]
Closing PR to test workflow

**Comment by gpmayorga** [comment]
Reopening again

**Comment by gpmayorga** [comment]
Closing PR to test workflow

**Comment by gpmayorga** [comment]
I was finally able to get everything to create and cleanup in the right order
https://github.com/centrifuge/website/actions/runs/3975491645

File: .github/workflows/website.yml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    paths-ignore:
+      - 'functions/**'      
+  pull_request:
+    paths-ignore:
+      - 'functions/**'
+      
+  workflow_dispatch:
+    inputs:
+      deploy_prod:
+        description: "Deploy to prod (requires approval and wait time 120m)"
```
Comment (Hornebom) [inline .github/workflows/website.yml:15]: Suggestion: If I remember it correctly the waiting time can be changed within the UI. Guess it makes more sense to document here where the waiting time is defined.

**Review by Hornebom — APPROVED** [review]

**Pull request merged**